tests/unit/validation/test_input_validation.py

```diff
@@ -270,52 +270,69 @@ def original_data() -> dict[ComponentType, np.ndarray]:
-    sym_power_sensor["id"] = [7, 8, 9, 10]
-    sym_power_sensor["measured_object"] = [12, 3, 13, 200]
-    sym_power_sensor["power_sigma"] = [1.0, np.nan, 0.0, -1.0]
-    sym_power_sensor["measured_terminal_type"] = [1, 1, 10, 1]
-
-    asym_power_sensor = initialize_array(DatasetType.input, ComponentType.asym_power_sensor, 4)
-    asym_power_sensor["id"] = [7, 8, 9, 10]
-    asym_power_sensor["measured_object"] = [12, 3, 13, 200]
-    asym_power_sensor["power_sigma"] = [1.0, np.nan, 0.0, -1.0]
-    asym_power_sensor["measured_terminal_type"] = [1, 1, 10, 1]
-
-    sym_current_sensor = initialize_array(DatasetType.input, ComponentType.sym_current_sensor, 4)
-    sym_current_sensor["id"] = [7, 8, 9, 10]
-    sym_current_sensor["measured_object"] = [12, 3, 13, 200]
-    sym_current_sensor["i_sigma"] = [1.0, np.nan, 0.0, -1.0]
-    sym_current_sensor["i_angle_sigma"] = [1.0, np.nan, 0.0, -1.0]
-    sym_current_sensor["measured_terminal_type"] = [1, 1, 10, 2]
-    sym_current_sensor["angle_measurement_type"] = [0, 1, 10, 2]
-
-    asym_current_sensor = initialize_array(DatasetType.input, ComponentType.asym_current_sensor, 4)
-    asym_current_sensor["id"] = [7, 8, 9, 10]
-    asym_current_sensor["measured_object"] = [12, 3, 13, 200]
-    asym_current_sensor["i_sigma"] = [1.0, np.nan, 0.0, -1.0]
-    asym_current_sensor["i_angle_sigma"] = [1.0, np.nan, 0.0, -1.0]
-    asym_current_sensor["measured_terminal_type"] = [1, 1, 10, 2]
-    asym_current_sensor["angle_measurement_type"] = [0, 1, 10, 2]
+    asym_voltage_sensor["u_sigma"] = [1.0, np.nan, 0.0, -1.0, np.inf]
+
+    sym_power_sensor = initialize_array(DatasetType.input, ComponentType.sym_power_sensor, 15)
+    sym_power_sensor["id"] = range(107, 107 + 15)
+    sym_power_sensor["measured_object"] = [12, 3, 13, 200, 3] + [3] * 10
+    sym_power_sensor["power_sigma"] = [1.0, np.nan, 0.0, -1.0, np.inf] * 2 + [np.nan] * 5
+    sym_power_sensor["measured_terminal_type"] = [1, 1, 10, 1, 1] + [1] * 10
+    sym_power_sensor["p_sigma"] = [np.nan] * 5 + [1.0, 1.0, 0.0, -1.0, np.inf] * 2
+    sym_power_sensor["q_sigma"] = [np.nan] * 5 + [1.0, np.inf, 0.0, -1.0, np.inf] * 2
+
+    asym_power_sensor = initialize_array(DatasetType.input, ComponentType.asym_power_sensor, 15)
+    asym_power_sensor["id"] = range(107, 107 + 15)
+    asym_power_sensor["measured_object"] = [12, 3, 13, 200, 3] + [3] * 10
+    asym_power_sensor["power_sigma"] = [1.0, np.nan, 0.0, -1.0, np.inf] * 2 + [np.nan] * 5
+    asym_power_sensor["measured_terminal_type"] = [1, 1, 10, 1, 1] + [1] * 10
+    asym_power_sensor["p_sigma"] = [[np.nan] * 3] * 5 + [
+        [1.0, 1.0, 1.0],
+        [np.inf, np.inf, np.inf],
+        [1.0, np.nan, 1.0],
+        [1.0, 1.0, -1.0],
+        [np.inf, 1.0, 1.0],
+    ] * 2
+    asym_power_sensor["q_sigma"] = [[np.nan] * 3] * 5 + [
+        [1.0, 1.0, 1.0],
+        [np.inf, np.inf, np.inf],
+        [np.inf, np.nan, 1.0],
+        [1.0, 1.0, -1.0],
+        [np.inf, 1.0, 1.0],
+    ] * 2
```

Comment:
additions needed to test the different options.

I deliberately added `100` to the original IDs so that the newly added IDs don't conflict with the existing IDs of other components. This decreases the minimal amount of changes in this PR